crates/anvil/src/eth/miner.rs

```diff
@@ -77,6 +81,11 @@ impl Miner {
         self.inner.wake();
     }
 
+    /// Get current mining mode
+    pub fn get_mining_mode(&self) -> RwLockReadGuard<'_, RawRwLock, MiningMode> {
+        self.mode.read()
+    }
```

Comment:
I'd like to return the converted `MiningModeKind` instead of the locked value

Reply:
The purpose of this get_mining_mode function is to provide access to the mine_mode without handling any conversion logic. 

The conversion to MiningModeKind should be handled by the external users, as needed. 

Let me know if you have any concerns with this approach.